src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const result = [];
+
+  for (const item of actions) {
+    switch (item.type) {
+      case 'addProperties':
+        newState = { ...newState, ...item.extraData };
+        result.push(newState);
```

Comment:
Instead of pushing the 'newState' to the 'result' array inside each case, do it once after the switch block to adhere to the DRY (Don't Repeat Yourself) principle.